catalog/transformer/catalog-transformer-geojson-input/src/main/java/ddf/catalog/transformer/input/geojson/GeoJsonInputTransformer.java

```diff
@@ -183,12 +163,40 @@ public Metacard transform(InputStream input, String id)
     }
 
     if (id != null) {
-      metacard.setId(id);
+      metacard.setAttribute(new AttributeImpl(Metacard.ID, id));
     }
 
     return metacard;
   }
 
+  protected Metacard getBaseMetacard(
+      Map<String, Object> properties, List<MetacardType> metacardTypes)
+      throws CatalogTransformerException {
+    final String propertyTypeName = (String) properties.get(METACARD_TYPE_PROPERTY_KEY);
+    MetacardImpl metacard;
+
+    if (isEmpty(propertyTypeName) || metacardTypes == null) {
+      LOGGER.debug(
+          "MetacardType specified in input is null or empty.  Assuming default MetacardType");
+      metacard = new MetacardImpl();
+    } else {
+      MetacardType metacardType =
+          metacardTypes
+              .stream()
+              .filter(type -> type.getName().equals(propertyTypeName))
+              .findFirst()
```

Comment:
❓ So searching for the metacard type based on the `METACARD_TYPE_PROPERTY_KEY` (line 175) doesn't accomplish all that we want for extensibility?

Reply:
I believe this was to handle the case where the override sets the metacard type to something not recognized by the local system. If that's not possible then this would be redundant.